src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -6,11 +6,20 @@
 #include <string.h>
 #include <vector>
 
-#include "config.h"
+#include "icushim.h"
 #include "locale.hpp"
 #include "holders.h"
 #include "errors.h"
 
+#ifndef FEATURE_FIXED_ICU_VERSION
+// The System.Globalization.Native.so is built in mode when it can use any supported version
+// of ICU installed on the target machine and not just the one it was built with. So we can 
+// only use stable APIs and the UDAT_STANDALONE_SHORTER_WEEKDAYS doesn't belong to those.
+// We also want to support ICU versions where the UDAT_STANDALONE_SHORTER_WEEKDAYS was not 
+// defined yet.
+#undef HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS
```

Comment:
> Does it make sense?

I think so.  However, if we make these changes, we should be able to remove the hack around `HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS` as well, correct.  When building in this new mode, we assert that the code will run on any version at least as high at the version you compiled on.  Since we only use stable stuff, we can use our configure tests.  Right?